liblwgeom/lwgeom_geos.c

```diff
@@ -443,19 +443,21 @@ LWGEOM2GEOS(const LWGEOM *lwgeom, int autofix)
 		ngeoms = lwc->ngeoms;
 		if ( ngeoms > 0 )
 			geoms = malloc(sizeof(GEOSGeom)*ngeoms);
-
+		j = 0;
+		// i is index of lwgeom geometry, j is index of geos geometry - skip empty collection entries
 		for (i=0; i<ngeoms; ++i)
 		{
+			if (lwgeom_is_empty(lwgeom)) continue; // skip empty
 			GEOSGeometry* g = LWGEOM2GEOS(lwc->geoms[i], 0);
 			if ( ! g )
 			{
 				while (i) GEOSGeom_destroy(geoms[--i]);
 				free(geoms);
 				return NULL;
 			}
-			geoms[i] = g;
+			geoms[++j] = g;
```

Comment:
Won't this leave `geoms[0]` null? You want to read-then-increment (`j++`) not increment-then-read (`++j`).